weed/filer/elastic/v7/elastic_store.go

```diff
@@ -0,0 +1,308 @@
+package elastic
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/chrislusf/seaweedfs/weed/filer"
+	"github.com/chrislusf/seaweedfs/weed/glog"
+	"github.com/chrislusf/seaweedfs/weed/pb/filer_pb"
+	weed_util "github.com/chrislusf/seaweedfs/weed/util"
+	jsoniter "github.com/json-iterator/go"
+	elastic "github.com/olivere/elastic/v7"
+)
+
+var (
+	indexType   = "_doc"
+	indexPrefix = ".seaweedfs_"
+)
+
+type ESEntry struct {
+	ParentId string `json:"ParentId"`
+	Entry    *filer.Entry
+}
+
+func init() {
+	filer.Stores = append(filer.Stores, &ElasticStore{})
+}
+
+type ElasticStore struct {
+	client      *elastic.Client
+	maxPageSize int
+}
+
+func (store *ElasticStore) GetName() string {
+	return "elastic7"
+}
+func (store *ElasticStore) Initialize(configuration weed_util.Configuration, prefix string) (err error) {
+	servers := configuration.GetString(prefix + "servers")
+	if servers == "" {
+		return fmt.Errorf("error elastic endpoints.")
+	}
+	store.maxPageSize = configuration.GetInt(prefix + "index.max_result_window")
+	if store.maxPageSize <= 0 {
+		return fmt.Errorf("error elastic index.max_result_window.")
+	}
+	glog.Infof("filer store elastic endpoints: %s, index.max_result_window:%d", servers, store.maxPageSize)
+	store.client, err = elastic.NewClient(
+		elastic.SetSniff(false),
+		elastic.SetHealthcheck(false),
+		elastic.SetURL(servers),
+	)
+	if err != nil {
+		return fmt.Errorf("init elastic %s: %v.", servers, err)
+	}
+	return nil
+}
+func (store *ElasticStore) BeginTransaction(ctx context.Context) (context.Context, error) {
+	return ctx, nil
+}
+func (store *ElasticStore) CommitTransaction(ctx context.Context) error {
+	return nil
+}
+func (store *ElasticStore) RollbackTransaction(ctx context.Context) error {
+	return nil
+}
+func (store *ElasticStore) KvDelete(ctx context.Context, key []byte) (err error) {
+	return filer.ErrKvNotImplemented
```

Comment:
could you please help to implement the kv operations? I do not have an ES cluster to develop.

It will be used to for more advanced features.

Reply:
of course, we will implement kv ops, but maybe next submits